### DISCUSSION TITLE
part-ii/2015-93/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#284] [Question comments]
Part II 2015 Number Theory: Paper 1, Section I, H 

https://questions.tripos.org/part-ii/2015-93/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

The first part is bookwork, but I'll type it up anyway. 

The Legendre symbol $(\frac{a}{p})$, for $p$ an odd prime and $a \in \mathbb{Z}$, is defined to be $a$ is a quadratic residue $\mod p$, to be $-1$ if $a$ is a quadratic non-residue $\mod p$, and to be $0$ if $p$ divides $a$.

Euler's criterion states that $(\frac{a}{p}) \equiv a^{\frac{p-1}{2}} \pmod{p}$. The result being clear for $a$ divisible by $p$ (both sides are equal to $0$), we assume that $(a,p) = 1$. Now by Fermat's little theorem, $a^{p-1} \equiv 1 \pmod{p}$, so $a^{\frac{p-1}{2}} \equiv \pm 1 \pmod{p}$. Consider the equation $x^{\frac{p-1}{2}} - 1 \equiv 0 \pmod{p}$. Any $x$ for which $x \equiv y^2$ is a solution (since then $x^{\frac{p-1}{2}} \equiv y^{p-1} \equiv 1 \pmod {p}$ by Fermat's little theorem), so since there are $\frac{p-1}{2}$ quadratic residues $\mod p$, we have $\frac{p-1}{2}$ solutions.

However, by Lagrange, this equation can only have $\frac{p-1}{2}$ solutions, so the quadratic residues are the only solutions, and therefore the quadratic non-residues $a$ must have $a^{\frac{p-1}{2}} \equiv -1 \pmod{p}$. But putting these results together gives Euler's criterion.

Now suppose $p$ is a prime dividing $n^2 + 4$. Since $n$ is odd, we can assume $p$ is odd. Then $n^2 \equiv -4 \pmod{p}$, so $(\frac{-4}{p}) = +1$, so $(\frac{-1}{p}) = + 1$ (taking out a square factor). By Euler's criterion, we get that $p \equiv 1 \pmod{4}$. Therefore all the prime factors of $n^2 + 4$ must be congruent to either $1$ or $5$ $\pmod{8}$.

Now if $n$ is odd, then we can write $n^2 + 4 = (2k + 1)^2 + 4 = 4k(k+1) + 5$, so since one of $k$ and $k+1$ must be even, we get that $n^2 + 4 \equiv 5 \pmod{8}$. The product of any set of integers congruent to $1$ $\pmod{8}$ is also congruent to $1$ $\pmod{8}$, so $n^2 + 4$ must therefore have a prime factor $p$ with $p \equiv 5 \pmod{8}$. Suppose there are only finitely many such primes. Let $n$ be their product. Then $n^2 + 4$ is not divisible by any prime congruent to $5$ $\pmod{8}$, but must have a prime factor of that form, so we have a contradiction. Therefore there must be infinitely many primes of this form.

</details>
